docs/proposals/control-data-plane-split/README.md

```diff
@@ -0,0 +1,260 @@
+# Proposal-1508: Separation of control and data plane
+
+This document proposes a design for separating the control and data planes.
+
+- Issue https://github.com/nginxinc/nginx-gateway-fabric/issues/1508
+- Status: Implementable
+
+## Background
+
+NGF composes its control and data plane containers into a single Kubernetes Pod. The control plane uses OS signals and a
+shared file system to configure and reload nginx. This architecture is problematic because the same RBAC policies govern
+the control and data planes and share CVE potential. A compromised control plane may impact the customer’s traffic. The
+Kubernetes API server may be affected if the data plane is compromised. In addition to security concerns, this
+architecture does not allow the control plane and data plane to scale independently.
+
+An added benefit is that this architecture makes it possible for the control plane to provision the data plane, unlocking the ability to support multiple Gateways with a single control plane.
+
+## Goals
+
+- Data plane and control plane containers run in separate Pods
+- The communication channel between the control and data planes can be encrypted
+- Data planes can register with control plane
+- Data plane can scale independently of the control plane
+- RBAC policy for data plane follows the principle of least privilege. The data plane should not have access to the
```

Comment:
what is included and excluded in the principle of least privilege when talking in terms for data plane and control plane

Reply:
This basically just means that the RBAC policy should only allow permissions that are absolutely needed for the data plane to function, nothing more. Same for control plane.